packages/types/src/video.ts

```diff
@@ -0,0 +1,60 @@
+// Progress callback called with percentage `0 <= p <= 1` of compression progress.
+type OnProgress = (p: number) => void
+
+export type ProcessOptions = {
+  videoName: string
+  compressedVideoName: string
+  videoCompression: number
+  chaptersConfig?: string
+  onProgress?: OnProgress
+  outputOptions?: string[]
+  videoFilters?: string
+}
+
+export type WriteVideoFrame = (data: Buffer) => void
+
+export type VideoRecording = {
+  api: RunModeVideoApi
+  controller?: BrowserVideoController
+}
+
+/**
+ * Interface yielded by the browser to control video recording.
+ */
+export type BrowserVideoController = {
+  /**
+   * A function that resolves once the video is fully captured and flushed to disk.
+   */
+  endVideoCapture: () => Promise<void>
+  /**
+   * Timestamp of when the video capture started - used for chapter timestamps.
+   */
+  startedVideoCapture: Date
+  postProcessFfmpegOptions?: Partial<ProcessOptions>
+  /**
+   * Used in single-tab mode to restart the video capture to a new file without relaunching the browser.
+   */
+  restart: () => Promise<void>
+  writeVideoFrame: WriteVideoFrame
+}
+
+/**
+ * Interface passed to the browser to enable capturing video.
+ */
+export type RunModeVideoApi = {
+  onError: (err: Error) => void
+  videoName: string
+  compressedVideoName: string
+  /**
+   * Create+use a new VideoController that uses ffmpeg to stream frames from `writeVideoFrame` to disk.
+   */
+  newFfmpegVideoController: (opts?: { webmInput?: boolean}) => Promise<BrowserVideoController>
```

Comment:
This does stand out a bit. Using `new` here makes me think of instantiation, but users of this API shouldn't really care much about that level of granularity. It might help to rename this to `setFfmpegVideoController()` to better indicate that there's more than just instantiation going on? I guess it could be unintuitive to return a Promise from what looks like a setter....maybe `useFfmpegVideoController()`? Just spitballing here.

Doing so might also help bring some parity to the `setVideoController`, which seems mutually exclusive with `newFfmpegVideoController`. Could similarly rename it to `useVideoController` and match the return values so that it's more obvious that one or the other API should be used.

Reply:
I like that, I made the change in [45e9b8f](https://github.com/cypress-io/cypress/pull/23579/commits/45e9b8f8e04b075f37966ef157f6d8b2f7020a2c)